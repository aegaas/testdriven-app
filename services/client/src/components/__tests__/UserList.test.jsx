import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import UserList from '../UserList';

const users = [
    {
        id: 1,
        username: 'zoey',
        email: 'z@aaronegaas.com',
        active: true
    },
    {
        id: 2,
        username: 'dexter',
        email: 'd@aaronegaas.com',
        active: true
    }
];

test('UserList renders properly', () => {
    const wrapper = shallow(<UserList users={users}/>);
    const element = wrapper.find('h4');
    expect(element.length).toBe(2);
    expect(element.get(0).props.className).toBe('well')
    expect(element.get(0).props.children).toBe('zoey')
});

test('UserList renders a snapshot properly', () => {
    const tree = renderer.create(<UserList users={users}/>).toJSON();
    expect(tree).toMatchSnapshot();
});
