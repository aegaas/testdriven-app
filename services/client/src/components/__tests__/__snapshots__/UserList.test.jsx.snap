// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList renders a snapshot properly 1`] = `
<div>
  <h4
    className="card card-body bg-light"
  >
    zoey
  </h4>
  <h4
    className="card card-body bg-light"
  >
    dexter
  </h4>
</div>
`;
